client.go

```diff
@@ -265,7 +266,7 @@ func (c *Client) Close() error {
 // By deafult, max retry is set to 25 and timeout is set to 30 minutes.
 //
 // If no ProcessAt or ProcessIn options are provided, the task will be pending immediately.
-func (c *Client) Enqueue(task *Task, opts ...Option) (*TaskInfo, error) {
+func (c *Client) Enqueue(ctx context.Context, task *Task, opts ...Option) (*TaskInfo, error) {
```

Comment:
I'm not against it, but I'm not sure if we want to make this API change in this PR. 
Unless you have a strong argument for this, I'll make internal changes without changing the public API of the package.